sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_fl.py

```diff
@@ -0,0 +1,254 @@
+import pytest
+from pathlib import Path
+from devtools_testutils import AzureRecordedTestCase, is_live
+from azure.core.exceptions import ResourceNotFoundError
+from .._util import _DSL_TIMEOUT_SECOND
+from test_utilities.utils import _PYTEST_TIMEOUT_METHOD
+from azure.ai.ml import Input, load_component
+from typing import Callable
+from azure.ai.ml.dsl._fl_scatter_gather_node import fl_scatter_gather
+from azure.ai.ml.entities._assets.federated_learning_silo import FederatedLearningSilo
+from azure.ai.ml.constants import AssetTypes
+from azure.ai.ml.entities._credentials import (
+    IdentityConfiguration,
+    IdentityConfigurationType,
+    ManagedIdentityConfiguration,
+)
+from azure.ai.ml.dsl import pipeline
+import os
+from azure.ai.ml import (
+    MLClient,
+)
+
+
+# RESOURCE NOTES - As an e2e test file, this file makes and references real AML resources.
+# All resources references by this file are contained within the following:
+# subscription: data science VM Team (DSVM)
+# resource group: fl-e2e-testing-rg
+# workspace: fl-e2e-testing-ws
+# If you're running this test locally in live mode, make sure you've set the above values in
+# your .env file.
+# resources of note:
+# - multiple storage containers
+# - multiple computes with varying managed identities to allow access to different storage containers
+@pytest.mark.timeout(timeout=_DSL_TIMEOUT_SECOND, method=_PYTEST_TIMEOUT_METHOD)
+@pytest.mark.e2etest
+@pytest.mark.core_sdk_test
+class TestDSLPipeline(AzureRecordedTestCase):
+    # DEV NOTE: This test expects a lot of resources to already exist in AML. The initial implementation
+    # of this test implemented all of these thigns under the 'fl-e2e-testing-rg' resource group and
+    # the 'fl-e2e-testing-ws' workspace inside that RG.
+    # The needed resources are:
+    # - In the workspace's storage account, 3 containers named 'silo-container1', 'silo-container23',
+    #       and 'agg-container'. Each container needs to have assigned the blob owner permission
+    #       to the 'test-msi-id' managed identity
+    # - 4 compute clusters in the workspace named siloCompute1/2/3 and aggCompute. Each of which
+    #       has been assigned the 'test-msi-id' ID
+    # - 4 datastores connected to the aforementioned storage containers named silo_datastore1/2/3 and
+    #       agg_datastore.
+    @pytest.mark.skipif(
+        condition=not is_live(),
+        reason=(
+            "TODO (2235034) The critical call to `client.jobs.create_or_update` seems to make different"
+            + "API calls in playback mode compared to recording mode"
+        ),
+    )
+    def test_fl_pipeline(
```

Comment:
This test looks extremely big to me. Is it possible to define some resources with YAML in test_configs, so that this test shall be shorter and cleaner?